### PR TITLE
python312Packages.citeproc-py: Remove nose dependency and re-enable tests

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Part of #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/citeproc-py/default.nix [hunk 1]
```diff
@@ -2,39 +2,41 @@
...
+    pytestCheckHook
   ];
-  checkPhase = "nosetests tests";
-  doCheck = false; # seems to want a Git repository, but fetchgit with leaveDotGit also fails
+
   pythonImportsCheck = [ "citeproc" ];
 
-  meta = with lib; {
+  meta = {
     homepage = "https://github.com/brechtm/citeproc-py";
```
Comment (emilazy) [inline pkgs/development/python-modules/citeproc-py/default.nix:36]: This has moved to https://github.com/citeproc-py/citeproc-py.

**Review by emilazy — CHANGES_REQUESTED** [review]
Looks like this project [might actually be becoming active again](https://github.com/citeproc-py/citeproc-py/issues/136). They’ve moved to pynose upstream, but just as a test runner, so using `pytestCheckHook` should be fine.

There have been a bunch of fix commits since the last release, so maybe we should pin an unstable version until they figure out doing releases again.

**Comment by emilazy** [comment]
(Sorry about all the nitpicking! I really appreciate everyone taking the initiative to deal with this massive list of packages; I just want to make sure we do what we can to get packages in a good state or else get rid of unnecessary burdens in the process.)

**Comment by pyrox0** [comment]
No worries! I appreciate the nitpicks, they help improve future packages I submit and/or update!

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 330237` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.citeproc-py</li>
    <li>python311Packages.citeproc-py.dist</li>
    <li>python311Packages.duecredit</li>
    <li>python311Packages.duecredit.dist</li>
    <li>python312Packages.citeproc-py</li>
    <li>python312Packages.citeproc-py.dist</li>
    <li>python312Packages.duecredit</li>
    <li>python312Packages.duecredit.dist</li>
  </ul>
</details>

**Pull request merged**